### ISSUE TITLE
Support for react-native/non-node environments

### ISSUE THREAD

**Issue by DT-1236** [jpeg-js/jpeg-js#92]
Are there plans to support this package in a non-node environment? I was able to get this working in react-native by adding the "buffer" package, but I'm not sure if there's a good way to conditionally add dependencies based on environment

**Comment by patrickhulce** [comment]
> Are there plans to support this package in a non-node environment?

This already works in the non-node environments with `useTArray` option.

> I was able to get this working in react-native by adding the "buffer" package

FWIW, you don't need to do this if you're comfortable working with `Uint8Array` output instead of a `Buffer`.

Happy to accept a PR with a "Working in Non-Node Environments" example pointing this out to the README.

**Comment by DT-1236** [comment]
Ah, yes, it _does_ work in react-native simply by adding the option. Thank you very much for your fast response. Any thoughts on adjusting the catch block for that section with something like this?
```
    if (err instanceof ReferenceError) {
      if (err.message === "Buffer is not defined") {
        err.message += " in this environment. Consider setting useTArray to true."
      }
    } 
    
    throw err;
```

Throwing an entirely new error might be more stylistically in-line with everything else

**Comment by DT-1236** [comment]
> (I'd even be up for automatically enabling useTArray if Buffer is not defined)

Hmm, that could potentially result in an unexpected return type. Granted I experienced that the other way around where I was trying to concat a `Buffer` onto an array 😆

I don't think there's going to be an easy way to get `encoder` to work in a non-node environment though

**Comment by patrickhulce** [comment]
thanks @DT-1236 ! we'll consider this closed by #93
